src/common/lspLanguageFeatures.ts

```diff
@@ -105,7 +106,10 @@ export class DiagnosticsAdapter<T extends ILanguageWorkerWithDiagnostics> {
 			.then((worker) => {
 				return worker.doValidation(resource.toString());
 			})
-			.then((diagnostics) => {
+			.then((allDiagnostics) => {
+				const diagnostics = allDiagnostics
+						.filter(diagnostic => (this._diagnosticCodesToIgnore || []).indexOf(diagnostic.code) === -1);
```

Comment:
I would suggest moving this check `this._diagnosticCodesToIgnore || []` to the constructor (since it's the unique place where it can be assigned).
Hence, this line can be:
```const diagnostics = allDiagnositcs.filter(diagnostic => diagnosticCodesToIgnore.includes(diagnostic.code));```

Reply:
in this case we want to exclude all the ones that I want to ignore. `includes` does something different, right? :p

Reply:
example: 
![image](https://user-images.githubusercontent.com/1200312/179806275-31ee3df7-d8cd-4580-8985-ad8aecfdbcf8.png)